web/courses.go

```diff
@@ -533,6 +539,61 @@ func (s *AutograderService) setLastApprovedAssignment(submission *pb.Submission,
 	return s.db.UpdateEnrollment(query)
 }
 
+// acceptRepositoryInvites tries to accept repository invitations for the given repository types on behalf of the given user.
+func (s *AutograderService) acceptRepositoryInvites(ctx context.Context, sc scm.SCM, user *pb.User, course *pb.Course, repoTypes []pb.Repository_Type) {
+	invites := []*github.RepositoryInvitation{}
+	for _, repoType := range repoTypes {
+		repo, err := s.db.GetRepositoryByType(user.ID, course.OrganizationID, repoType)
+		if err != nil {
+			s.logger.Error("Failed to get repo: %s", err)
+			continue
+		}
+		// TODO: pb.Repository should contain Path
+		path := strings.Split(repo.GetHTMLURL(), "/")
+
+		repoInvites, err := sc.GetRepositoryInvites(ctx, &scm.RepositoryOptions{ID: repo.RepositoryID, Owner: course.OrganizationPath, Path: path[len(path)-1]})
+		if err != nil {
+			s.logger.Errorf("Failed to get repository invites for %v: %s - %s", repo.RepoType, user.Login, err)
+			continue
+		}
+		for _, invite := range repoInvites {
+			// Some repositories contain more than one invite
+			// and we only want to accept the invite for the
+			// user we are enrolling.
+			if *invite.Invitee.Login != user.Login {
+				continue
+			}
+			invites = append(invites, invite)
+		}
+	}
+
+	if !(len(invites) > 0) {
+		return
+	}
+
+	user, err := s.db.GetUser(user.ID)
+	if err != nil {
+		s.logger.Errorf("Failed to get user %d: %v", user.ID, err)
+		return
+	}
+	// We need the user access token to accept invitations on their behalf
+	accessToken, err := user.GetAccessToken("github")
+	if err != nil {
+		fmt.Errorf("Failed to get access token for user %s: %v", user.Login, err)
+		return
+	}
+	sc, ok := s.scms.GetSCM(accessToken)
+	if !ok {
+		s.logger.Errorf("GetSCM failed: could not get SCM for user: %s", user.Login)
+		return
+	}
+	for _, invite := range invites {
+		if err := sc.AcceptRepositoryInvite(ctx, &scm.RepositoryInvitationOptions{InvitationID: uint64(*invite.ID)}); err != nil {
```

Comment:
Again, assuming the rename in `scm.go` and the above tweaks, you can do this directly:
```go
		if err := sc.AcceptRepositoryInvite(ctx, invite); err != nil {
```